modules: hal: infineon: Add Kconfig option for pullups in Wifi Host Driver

MagpieEmbedded opened the pull request:
This option allows board configurations to specify how to set the SDIO pull ups when the SPI bus goes to sleep, using the Infineon WHD. This is a board specific value.

This PR is dependent on a PR for `hal_infineon`: https://github.com/zephyrproject-rtos/hal_infineon/pull/28

It provides a fix to the issue highlighted in #90793 and #91226 

The default configuration is to leave things as they are for other boards relying on the current behaviour.

Happy to make further changes as necessary.

Thanks, Tim 👍

MagpieEmbedded committed:
modules: hal: infineon: Add Kconfig option for pullups in Wifi Host Driver

This option allows board configurations to specify how to set the SDIO
pull ups when the SPI bus goes to sleep, using the Infineon WHD.
This is a board specific value.

Signed-off-by: Magpie Embedded <magpieembedded@gmail.com>

MagpieEmbedded commented on drivers/wifi/infineon/Kconfig.airoc:
```diff
@@ -59,6 +59,13 @@ config WHD_WIFI_COUNTRY
 
+config WHD_DISABLE_SDIO_PULLUP_DURING_SPI_SLEEP
+	bool "Disable SDIO pullups during SPI sleep"
+	default n
```
Done

nordicjm reviewed (dismissed):
Kconfig/cmake changes OK

MagpieEmbedded commented:
Updated to refer to the updated `hal_infineon` module which supports this PR and rebased on top of `main`. @nordicjm Could you please take a look when you can? Thanks, Tim

MagpieEmbedded commented:
This should be ready to merge if someone can remove the DNM tag? Thanks

MagpieEmbedded commented:
This has been rebased onto main last night, should be ready for merge 👍

dkalowsk commented:
moving this to 4.3 as it's not clear the rebase will be fixed for the 4.2 tagging, and it's not clear this is a bug fix at this point.

MagpieEmbedded commented:
Hi @dkalowsk this has been rebased now, should be working. This is a new feature, not a bug fix. It will enable the necessary fixes to PR #90793 which provides networking support for the Pico 2W. Depending on the timeline, that could be included in the 4.2 release. Happy to work with you on this or leave it until the next release. Thanks, Tim

bricle commented:
@jukkar  Could you please re-visit this PR? Many thanks.

ThreeEights reviewed (approved):
This is needed for Pico 2 W support.
